client/shipping-label/views/dialog.js

```diff
@@ -2,90 +2,116 @@ import React from 'react';
 import Dialog from 'components/dialog';
 import ActionButtons from 'components/action-buttons';
 import { translate as __ } from 'lib/mixins/i18n';
-import sum from 'lodash/sum';
 import AddressStep from './steps/address';
 import PackagesStep from './steps/packages';
 import RatesStep from './steps/rates';
 import PreviewStep from './steps/preview';
 import { hasNonEmptyLeaves } from 'lib/utils/tree';
 import { sprintf } from 'sprintf-js';
+import isEqual from 'lodash/isEqual';
+import debounce from 'lodash/debounce';
 
-const PrintLabelDialog = ( props ) => {
-	const currencySymbol = props.storeOptions.currency_symbol;
+class PrintLabelDialog extends React.Component {
+	maybeGetNewRates( nextProps ) {
+		const useNormalizedOrigin = nextProps.form.origin.selectNormalized;
+		const useNormalizedDestination = nextProps.form.destination.selectNormalized;
 
-	const getTotalCost = () => {
-		const ratesInfo = props.form.rates;
-		const ratesCost = ratesInfo.values.map( ( rateId, index ) => ratesInfo.available[ index ][ rateId ].rate );
-		return sum( ratesCost ).toFixed( 2 );
-	};
+		// Determine if we need to compare user-entered values, or normalized ones
```

Comment:
This component is "too smart". It's better to have as much logic as possible in one place, and in this case the place should be `actions.js`. The `props` you are checking are part of the `state`, so they are available for the `actions`.

A good way of knowing that something shouldn't be in the component, is that you needed to change from a functional component to a Class component just for this logic.